src/Switch/index.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { forwardRef } from 'react';
+import styled from 'styled-components';
+import { CustomCheckboxContainer, CustomCheckboxInput } from '@reach/checkbox';
+import { MinervaProps, Box, systemProps } from '../layout';
+import VisuallyHidden from '../VisuallyHidden';
+// import { useTheme } from '../theme';
+
+const SwitchContainer = styled(Box)(
+  {
+    position: 'relative',
+    width: '75px',
+    display: 'inline-block',
+    verticalAlign: 'middle',
+    textAlign: 'left',
+  },
+  systemProps
+);
+
+export type ControlBoxProps = {
+  checked?: boolean;
+  theme?: any;
+};
+
+const ControlBox = styled(Box)<ControlBoxProps>(
+  (props: ControlBoxProps) => ({
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderStyle: 'solid',
+    borderWidth: '1px',
+    borderRadius: '4px',
+    width: '16px',
+    height: '16px',
+    bg: 'primary',
+    marginRight: '8px',
+    transition: 'background-color 120ms ease 0s, box-shadow 250ms ease 0s',
+    borderColor: props.checked ? '#fff' : '#ecebed',
+    backgroundImage: `url("data:image/svg+xml;charset=utf-8,%3Csvg viewBox='0 0 16 16' fill='%23fff' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M5.707 7.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4a1 1 0 00-1.414-1.414L7 8.586 5.707 7.293z'/%3E%3C/svg%3E")`,
+    backgroundColor: props.checked ? '#5850ec' : '#fff',
+    ':focus': {
+      borderColor: '#a4cafe',
+      boxShadow: '0 0 0 3px rgba(118,169,250,.45)',
+      outline: 0,
+    },
+  }),
+  systemProps
+);
+
+const SwitchLabel = styled.label`
+  display: block;
+  overflow: hidden;
+  cursor: pointer;
+  border: 0 solid #ccc;
+  border-radius: 20px;
+  margin: 0;
+`;
+
+const SwitchInner = styled.span`
+  display: block;
+  width: 200%;
+  margin-left: -100%;
```

Comment:
Usually for an animation like this, using the margin to transform it can be unpredictable in some browsers and sometimes have weird side effects.

I'd recommend using the `transform` CSS property and using `translateX` to move the inner switch toggle

```suggestion
transform: ${props => props.checked ? 'translateX(100%)' : 'translateX(0%)'}
```